lib/logstash/filters/translate.rb

```diff
@@ -137,18 +137,18 @@ def load_yaml(registering=false)
 
     begin
       @dictionary.merge!(YAML.load_file(@dictionary_path))
-    rescue Exception => e
```

Comment:
There is an understandable but flawed reason why this line was added.  It is because YAML (on jruby, == Psych) defines SyntaxError as a subclass of RuntimeError not StandardError, so hence the `rescue Exception => e`

This is a classic case of the discussion that @jsvd and I raised about plugins handling specific errors in their rescue blocks. See [this discussion issue](https://github.com/elastic/logstash/issues/2477)

The work that may raise an Error in line 139 is doing many things, opening and parsing a file then merging the results to a Hash.

There should be two rescue sections.
```ruby
rescue YAML::SyntaxError => e
  msg = "..." #specific to the syntax problem
  raise Exception.new(msg) #this will be thrown up to the pipeline; v2.0 should rescue, attempt to log it and reraise, this will kill the worker thread
rescue StandardError => e
  msg = "..." #general message
  #log error
```

Also on JRUBY a "file not found" will raise Errno::ENOENT and this happens before Psych tries to parse anything from the file.  Consider trapping this as a specific rescue and creating a more helpful message to log.